rest/config_flags.go

```diff
@@ -0,0 +1,199 @@
+
+		"unsupported.stats_log_frequency": {&config.Unsupported.StatsLogFrequency, fs.String("unsupported.stats_log_frequency", "", "How often should stats be written to stats logs")},
+		"unsupported.use_stdlib_json":     {&config.Unsupported.UseStdlibJSON, fs.Bool("unsupported.use_stdlib_json", false, "Bypass the jsoniter package and use Go's stdlib instead")},
+
+		"unsupported.http2.enabled": {&config.Unsupported.HTTP2.Enabled, fs.Bool("unsupported.http2.enabled", false, "Whether HTTP2 support is enabled")},
+
+		"max_file_descriptors": {&config.MaxFileDescriptors, fs.Uint64("max_file_descriptors", 0, "Max # of open file descriptors (RLIMIT_NOFILE)")},
+	}
+}
+
+// fillConfigWithFlags fills in the config values from setConfigFlags if the user
+// has explicitly set the flags
+func fillConfigWithFlags(fs *flag.FlagSet, flags map[string]configFlag) (errorMessages error) {
+	fs.Visit(func(f *flag.Flag) {
+		if val, exists := flags[f.Name]; exists {
+			rval := reflect.ValueOf(val.config).Elem()
+
+			pointer := true // Distinguish if to use rval.Set or *val.config
+			// Convert to pointer if not already
+			if rval.Kind() != reflect.Ptr && rval.CanAddr() {
+				rval = rval.Addr()
+				pointer = false
+			}
+
+			switch rval.Interface().(type) {
+			case *string:
+				*val.config.(*string) = *val.flagValue.(*string)
+			case *[]string:
+				list := strings.Split(*val.flagValue.(*string), ",")
+				*val.config.(*[]string) = list
+			case *uint:
+				*val.config.(*uint) = *val.flagValue.(*uint)
+			case *uint64:
+				*val.config.(*uint64) = *val.flagValue.(*uint64)
+			case *int:
+				if pointer {
+					rval.Set(reflect.ValueOf(val.flagValue))
+				} else {
+					*val.config.(*int) = *val.flagValue.(*int)
+				}
+			case *bool:
+				rval.Set(reflect.ValueOf(val.flagValue))
+			case *base.ConfigDuration:
+				duration, err := time.ParseDuration(*val.flagValue.(*string))
+				if err != nil {
+					err = fmt.Errorf("flag %s error: %w", f.Name, err)
+					errorMessages = multierror.Append(errorMessages, err)
+					return
+				}
+				if pointer {
+					rval.Set(reflect.ValueOf(&base.ConfigDuration{Duration: duration}))
+				} else {
+					*val.config.(*base.ConfigDuration) = base.ConfigDuration{Duration: duration}
+				}
+			case *base.RedactionLevel:
+				var rl base.RedactionLevel
+				err := rl.UnmarshalText([]byte(*val.flagValue.(*string)))
+				if err != nil {
+					err = fmt.Errorf("flag %s error: %w", f.Name, err)
+					errorMessages = multierror.Append(errorMessages, err)
+					return
+				}
+				*val.config.(*base.RedactionLevel) = rl
+			case *base.LogLevel:
+				var ll base.LogLevel
+				err := ll.UnmarshalText([]byte(*val.flagValue.(*string)))
+				if err != nil {
+					err = fmt.Errorf("flag %s error: %w", f.Name, err)
+					errorMessages = multierror.Append(errorMessages, err)
+					return
+				}
+				rval.Set(reflect.ValueOf(&ll))
+			default:
+				panic(fmt.Sprintf("Error: Unknown type %v for flag %v\n", rval.Type(), f.Name))
```

Comment:
I guess we can do this actually. Didn't see this function was returning errors previously.

```suggestion
				errorMessages = multierror.Append(errorMessages, fmt.Errorf("Unknown type %v for flag %v\n", rval.Type(), f.Name))
```

Reply:
Leaving as a panic as it should never get to that stage for a user (only a developer when adding new config options but not adding a case for a new/different type). 

Let me know if you still think I should change it to just an error.